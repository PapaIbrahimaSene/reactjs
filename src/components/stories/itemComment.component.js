import React from 'react';
import moment from 'moment';
import renderHTML from 'react-render-html';
import { Feed, Icon } from 'semantic-ui-react';

export const ItemComment = ({
  comment: { by, score, text, time, comments }
}) => (
  <Feed>
    <Feed.Event>
      <Feed.Label>
        <Icon name='user circle' color='orange'></Icon>
      </Feed.Label>
      <Feed.Content style={{fontSize: '90%'}}>
        <Feed.Summary>
          by <a>{ by }</a>
          <Feed.Date>
            { moment(time * 1000).calendar() }
          </Feed.Date>
        </Feed.Summary>
        <Feed.Extra text>
          { renderHTML(String(text)) }
        </Feed.Extra>
        <Feed.Extra text>
          { comments && comments.length > 0  && comments.map(comment => (
            <ItemComment key={ comment.id } comment={ comment }></ItemComment>
          ))}
        </Feed.Extra>
      </Feed.Content>
    </Feed.Event>
  </Feed>
);