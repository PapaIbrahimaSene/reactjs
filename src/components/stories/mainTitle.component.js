import React from 'react';
import { Container, Header, Icon } from 'semantic-ui-react'

export const MainTitle = () => (
    <Container text style={{ marginTop: '5em'}} >
    <Header as='h1' >
    <Icon inverted name = "hacker news" size = "big" color='red'/>
    News Stories and Comments From HackerNews
    <Header as='h6'>Top 10 News Stories and related - up to - 20 comments from Y Combinator's API</Header>
    </Header>
    </Container>
);

export default MainTitle;