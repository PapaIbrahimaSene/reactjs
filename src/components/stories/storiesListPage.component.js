import React from 'react';
import { MainMenu } from '../core/mainMenu.component';
import StoriesList from './storiesList.component';
import MainTitle from './mainTitle.component';

export const StoriesListPage = () => (
  <div>
    <MainMenu />
    <MainTitle />
    <StoriesList />
  </div>
);

export default StoriesListPage;