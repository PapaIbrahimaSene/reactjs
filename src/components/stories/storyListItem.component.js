import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Button, Feed, Icon, List } from 'semantic-ui-react';
import { ItemComment } from './itemComment.component';

const mapStateToProps = state => ({
  comment: state.comment
});

export class StoryListItem extends React.Component {
  state = {
    viewComment: false
  };

  handleToggleCommentsClick = () => {
    this.setState(prevState => Object.assign(prevState, {
      viewComment: !prevState.viewComment
    }));
  }

  render() {
    return (
      <List.Item>
        <List.Content floated='left'>
          <Feed>
            <Feed.Event>
              <Feed.Content>
                <Feed.Summary>
                  by <a>{ this.props.story.by }</a>
                  <Feed.Date>
                    { moment(this.props.story.time * 1000).calendar() }
                  </Feed.Date>
                </Feed.Summary>
                <Feed.Extra text >
                  <h4>
                    <a href={ this.props.story.url }
                      target='_blank'>
                      { this.props.story.title }
                    </a>
                  </h4>
                </Feed.Extra>
                <Feed.Meta>
                  <Feed.Like>
                  <Icon name='thumbs up outline' color='red' />
                  { this.props.story.score } Points &nbsp;&nbsp;&nbsp;
                  <Icon name='user circle' color='red' /> &nbsp;
                  { this.props.story.descendants || 'No'} Comments
                  </Feed.Like>
                </Feed.Meta>
              </Feed.Content>
            </Feed.Event>
          </Feed>
        </List.Content>
        <List.Content floated='right'>
            <Button
              basic color='red'
              style={{ fontSize: '80%', padding: '12px' }}
              disabled={ !this.props.story.descendants }
              onClick={ this.handleToggleCommentsClick }>
              { !this.state.viewComment && (<span>View Comments</span>) }
              { this.state.viewComment && (<span>Hide Comments</span>) }
              </Button>
        </List.Content>
        { this.state.viewComment && this.props.story.comments && this.props.story.comments.length > 0 &&
          <List.Content style={{ paddingTop: '12px' }} floated='left'>
            { this.props.story.comments.map(comment => (
              <ItemComment key={ comment.id } comment={ comment }></ItemComment>
            ))}
          </List.Content>
        }
      </List.Item>
    );
  }
}

export default connect(
  mapStateToProps
)(StoryListItem);
