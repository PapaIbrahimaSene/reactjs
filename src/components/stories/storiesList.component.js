import React from 'react';
import { connect } from 'react-redux';
import { StoryListItem } from './storyListItem.component';

import { fetchTopStories } from '../../selectors/stories.selector';

import {
  Container,
  Dimmer,
  List,
  Loader,
} from 'semantic-ui-react';

const mapDispatchToProps = (dispatch) => ({
  onLoadTopStories: () => dispatch(fetchTopStories())
});

const mapStateToProps = state => ({
  stories: state.stories,
});

export class StoriesList extends React.Component {
  state = {
    loading: true
  };

  componentDidMount() {
    this.props.onLoadTopStories()
      .then(() => this.setState(prevState => Object.assign(prevState, {
        loading: false
      })));
  }

  render() {
    return (
      <Container text style={{ marginTop: '5em' }}>
        { this.state.loading &&
          <Dimmer page active inverted>
            <Loader
              content='Loading'
              size='big' />
          </Dimmer>
        }
        <List relaxed='very' divided style={{ marginBottom: '24px' }}>
          {
            this.props.stories && this.props.stories.map(story => (
              <StoryListItem key={ story.id } story={ story } />
            ))
          }
        </List>
      </Container>
    );
  }
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StoriesList);
