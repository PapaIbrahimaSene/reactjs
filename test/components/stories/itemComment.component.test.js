import React from 'react';
import { shallow } from 'enzyme';
import { ItemComment } from '../../../src/components/stories/itemComment.component';
import stories from '../../fixtures/stories';

test('should create ItemComment', () => {
    const wrapper = shallow(<ItemComment comment={stories[0].comments[1]} />);

    expect(wrapper).toMatchSnapshot();
});