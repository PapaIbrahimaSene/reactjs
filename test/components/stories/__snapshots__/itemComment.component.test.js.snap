// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create ItemComment 1`] = `
<Feed>
  <FeedEvent>
    <FeedLabel>
      <Icon
        as="i"
        color="orange"
        name="user circle"
      />
    </FeedLabel>
    <FeedContent
      style={
        Object {
          "fontSize": "90%",
        }
      }
    >
      <FeedSummary>
        by 
        <a>
          zellyn
        </a>
        <FeedDate>
          Today at 11:46 AM
        </FeedDate>
      </FeedSummary>
      <FeedExtra
        text={true}
      >
        You can do merkle-tree-signing and other clever things with cryptography to provide auditability without needing any kind of distributed consensus via proof of work/stake/etc. so I think the flowchart could have a couple more "No" branches.
      </FeedExtra>
      <FeedExtra
        text={true}
      >
        <ItemComment
          comment={
            Object {
              "by": "pythonaut_16",
              "comments": Array [
                Object {
                  "by": "beaconstudios",
                  "comments": Array [
                    Object {
                      "by": "zellyn",
                      "id": 18544030,
                      "parent": 18543960,
                      "text": "<a href=\\"https:&#x2F;&#x2F;twitter.com&#x2F;whitequark&#x2F;status&#x2F;946886702932557824\\" rel=\\"nofollow\\">https:&#x2F;&#x2F;twitter.com&#x2F;whitequark&#x2F;status&#x2F;946886702932557824</a><p><a href=\\"https:&#x2F;&#x2F;twitter.com&#x2F;hikari_no_yume&#x2F;status&#x2F;933791331146784768\\" rel=\\"nofollow\\">https:&#x2F;&#x2F;twitter.com&#x2F;hikari_no_yume&#x2F;status&#x2F;933791331146784768</a><p>More seriously, some people apparently consider git a blockchain: <a href=\\"https:&#x2F;&#x2F;medium.com&#x2F;@shemnon&#x2F;is-a-git-repository-a-blockchain-35cb1cd2c491\\" rel=\\"nofollow\\">https:&#x2F;&#x2F;medium.com&#x2F;@shemnon&#x2F;is-a-git-repository-a-blockchain...</a>",
                      "time": 1543337898,
                      "type": "comment",
                    },
                  ],
                  "id": 18543960,
                  "kids": Array [],
                  "parent": 18543917,
                  "text": "what you&#x27;re describing is a Merkle tree. Blockchains are essentially Merkle trees with a distributed consensus algorithm. Take away the consensus algorithm and you&#x27;re back to Merkle trees.",
                  "time": 1543337577,
                  "type": "comment",
                },
              ],
              "id": 18543917,
              "kids": Array [],
              "parent": 18543884,
              "text": "Doesn&#x27;t that somewhat depend on how you define a blockchain? Blockchain doesn&#x27;t inherently require proof of work&#x2F;stake&#x2F;etc, just a chain where each block contains a hash of the previous block(s). Proof of work&#x2F;stake just happens to be a popular implementation detail because of cryptocurrencies.",
              "time": 1543337353,
              "type": "comment",
            }
          }
          key="18543917"
        />
      </FeedExtra>
    </FeedContent>
  </FeedEvent>
</Feed>
`;
